File: RenetikObjc/Library/Classes/Core/Categories/UIKit/UIWindow+CSExtension.m
//
// Created by Rene Dohan on 6/26/13.
//
// To change the template use AppCode | Preferences | File Templates.
//


#import "UIWindow+CSExtension.h"


@implementation UIWindow (CSExtension)

- (void)showRootController:(UIViewController *)controller {
    [UIView transitionWithView:self duration:0.5 options:UIViewAnimationOptionTransitionCrossDissolve
                    animations:^(void) {
                        BOOL oldState = [UIView areAnimationsEnabled];
                        [UIView setAnimationsEnabled:NO];
                        self.rootViewController = controller;
                        [UIView setAnimationsEnabled:oldState];
                    } completion:nil];
}

@end

File: Renetik/Renetik/Classes/Core/Extensions/UIKit/UIWindow+CSExtension.swift
//
// Created by Rene Dohan on 3/12/20.
//

import Foundation
import UIKit

public extension UIWindow {

    class func construct(_ controller: UIViewController) -> Self {
        let window: Self = self.construct()
        window.rootViewController = controller
        window.makeKeyAndVisible()
        return window
    }

    class var window: UIWindow { delegate.window! }
    class var safeWidth: CGFloat { window.width - (window.safeAreaInsets.left + window.safeAreaInsets.right) }
    class var safeHeight: CGFloat { window.height - (window.safeAreaInsets.top + window.safeAreaInsets.bottom) }

    func show(root controller: UIViewController) -> Self {
        UIView.transition(with: self, duration: 0.5, options: .transitionCrossDissolve, animations: {
            let oldState = UIView.areAnimationsEnabled
            UIView.setAnimationsEnabled(false)
            self.rootViewController = controller
            UIView.setAnimationsEnabled(oldState)
        })
        return self
    }


}